modules/cli/src/main/scala/scala/cli/commands/Bsp.scala

```diff
@@ -29,10 +30,27 @@ object Bsp extends ScalaCommand[BspOptions] {
       argsSeq => {
         val sharedOptions = getSharedOptions()
         sharedOptions.inputs(argsSeq, () => Inputs.default())
-          .map { i =>
+          .map { initialInputs =>
             if (sharedOptions.logging.verbosity >= 3)
-              pprint.err.log(i)
-            Build.updateInputs(i, buildOptions(sharedOptions))
+              pprint.err.log(initialInputs)
+
+            val buildOptions0    = buildOptions(sharedOptions)
+            val logger           = sharedOptions.logging.logger
+            val persistentLogger = new PersistentDiagnosticLogger(logger)
+
+            val allInputs: Inputs =
+              CrossSources.forInputs(
+                initialInputs,
+                Sources.defaultPreprocessors(
+                  buildOptions0.scriptOptions.codeWrapper.getOrElse(CustomCodeWrapper),
+                  buildOptions0.archiveCache,
+                  buildOptions0.internal.javaClassNameVersionOpt
+                ),
+                persistentLogger
+              ).left.map((_, Scope.Main))
+                .map(_._2).getOrElse(initialInputs)
```

Comment:
It seems we're discarding an error here (if what `forInputs` returns is left, the error on the left is simply discarded). Before the changes of this PR, maybe you should change `argsToInputs` to a `Seq[String] => Either[BuildException, Inputs]` (and change its use sites accordingly, so that they create / propagate `BuildException`s), so that if `forInputs` returns a `Left`, it gets passed to the caller of `argsToInputs`.

Reply:
I don't think we're discarding an error because it is processed in `BspImpl` and then reported in diagnostics.

If using directives contains errors then we should start `Bsp` with `initialInputs` which contains inputs only from args, therefore I added `.getOrElse(initialInputs)`